objects/SetupChecker/script.lua

```diff
@@ -1295,3 +1221,84 @@ function tFind(table, needle)
     end
     return nil
 end
+---
+
+--- Update the UI XML of the current object.
+-- This takes a list of function to use to update the XML table.
+-- Each function should take a table element, update it with any relevant changes
+-- and return `true` if updateXml should recurse into the children of the element.
+-- Note: The functions should be prepared to be called on child elements, even if
+-- recursion was not requested.
+-- @param updateFunctions The list of functions to use to update the table
+function updateXml(updateFunctions)
+    local function recurse(t)
+        local shouldRecurse = false
+        for _, f in pairs(updateFunctions) do
+            if f(t) then
+                shouldRecurse = true
```

Comment:
you should break after setting this to true

Reply:
No. There may be another function that wants to mutate the table in the list of update functions.

Reply:
oh shoot yea you're right